pallets/ajuna-awesome-avatars/src/migration/v6/mod.rs

```diff
@@ -0,0 +1,784 @@
+		let current_version = Pallet::<T>::in_code_storage_version();
+		let onchain_version = Pallet::<T>::on_chain_storage_version();
+
+		if onchain_version == 5 && current_version == 6 {
+			log::info!(
+				target: LOG_TARGET,
+				"Nothing to check in v5 -> v6 migration."
+			);
+		} else {
+			log::info!(
+				target: LOG_TARGET,
+				"Migration post upgrade v6 did not execute."
+			);
+		}
+
+		Ok(())
+	}
+}
+
+/// The code following here is for the multiblock migration.
+///
+/// **NOTE** We have multiple multiblock migrations and the order **matters**.
+/// Order:
+///    1. LazyMigrationPlayerSeasonConfigsV5ToV6
+///    2. LazyMigrationSeasonStatsV5ToV6
+///    3. LazyMigrationAvatarV5ToV6
+///    4. LazyTradeStatsMapCleanup
+///
+/// We have multiple multiblock migrations that are interdependent because one writes
+/// into the auxiliary storage `TradeStatsMap`, where a following migration reads from there, and we
+/// also have a cleanup migration, which removes all the remaining entries from the `TradeStatsMap`
+/// storage.
+///
+/// The code is inspired by the following example:
+/// https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/examples/multi-block-migrations/src/lib.rs
+pub mod mbm {
+	use super::*;
+
+	pub struct LazyMigrationPlayerSeasonConfigsV5ToV6<T: Config, W: weights::WeightInfo>(
+		PhantomData<(T, W)>,
+	);
+	impl<T: Config, W: weights::WeightInfo> SteppedMigration
+		for LazyMigrationPlayerSeasonConfigsV5ToV6<T, W>
+	{
+		type Cursor = (T::AccountId, SeasonId);
+		// Without the explicit length here the construction of the ID would not be infallible.
+		type Identifier = MigrationId<25>;
+
+		/// The identifier of this migration. Which should be globally unique.
+		fn id() -> Self::Identifier {
+			MigrationId { pallet_id: *b"aaa-player-season-configs", version_from: 5, version_to: 6 }
+		}
+
+		/// The actual logic of the migration.
+		///
+		/// This function is called repeatedly until it returns `Ok(None)`, indicating that the
+		/// migration is complete. Ideally, the migration should be designed in such a way that each
+		/// step consumes as much weight as possible. However, this is simplified to perform one
+		/// stored value mutation per block.
+		fn step(
+			mut cursor: Option<Self::Cursor>,
+			meter: &mut WeightMeter,
+		) -> Result<Option<Self::Cursor>, SteppedMigrationError> {
+			let required = W::player_season_configs_step();
+			// If there is not enough weight for a single step, return an error. This case can be
+			// problematic if it is the first migration that ran in this block. But there is nothing
+			// that we can do about it here.
+			if meter.remaining().any_lt(required) {
+				return Err(SteppedMigrationError::InsufficientWeight { required });
+			}
+
+			let mut migration_count = 0u32;
+			// We loop here to do as much progress as possible per step.
+			loop {
+				if meter.try_consume(required).is_err() {
```

Comment:
This loop was probably a bit annoying to write, or?  
I want to change this `step` function to already do the looping, that you can focus on the migration step itself.  

You think this would make it better or worse @clangenb ?

Reply:
I have actually been thinking about the same thing, I think it would improve things, and reduce the total lines of code written. :)